tests/api/test_api.py

```diff
@@ -105,25 +105,67 @@ def test_spot_detail_get_auth_error(self):
             response = client.get(url)
             self.assertStatusCode(url, response, 401)
 
+
     @mock.patch("api.serializers.SpotSerializer.determine_stadsdeel")
-    def test_spot_detail_post(self, determine_stadsdeel):
+    def test_spot_detail_spottypes_post(self, determine_stadsdeel):
         determine_stadsdeel.return_value = Spot.Stadsdelen.Centrum
-
         url = reverse("spot-list")
-        data = {
-            "locatie_id": "123",
-            "spot_type": Spot.SpotType.blackspot,
-            "description": "Test spot",
-            "point": '{"type": "Point","coordinates": [4.9239022,52.3875654]}',
-            "actiehouders": "Actiehouders test",
-            "status": "voorbereiding",
-            "jaar_blackspotlijst": 2019,
-        }
-        response = self.write_client.post(url, data=data)
-        self.assertStatusCode(url, response, expected_status=201)
+        #  the @pytest.mark.parametrize is not supported with unittests. The 
+        #  easiest and cleanest way is so create a list
+        #  of tuples and loop through it. The same what the parametrize 
+        #  decorator does. the tuple contains the data object but can also
+        #  contain the status code and other variables.
```

Comment:
Ook dit vind ik geen geschikte oplossing. Sowieso plaats je nu de dictionary met data in een set en die in een array. De set is overbodig. Maar daarnaast is dit nog steeds niet DRY. Het grootste deel van deze array overlapt. Dan heb ik nog liever dat we parametrize namaken op deze manier:

```
    def test_spot_detail_post(self):
        for spot_type, _label in Spot.SpotType.choices:
            self._test_spot_detail_post(spot_type=spot_type)

    @mock.patch("api.serializers.SpotSerializer.determine_stadsdeel")
    def _test_spot_detail_post(self, determine_stadsdeel, spot_type):
        determine_stadsdeel.return_value = Spot.Stadsdelen.Centrum

        url = reverse("spot-list")
        data = {
            "locatie_id": random.randint(1, 999999),
            "spot_type": spot_type,
            "description": "Test spot",
            "point": '{"type": "Point","coordinates": [4.9239022,52.3875654]}',
            "actiehouders": "Actiehouders test",
            "status": "voorbereiding",
        }

        if spot_type in [Spot.SpotType.blackspot, Spot.SpotType.wegvak]:
            data["jaar_blackspotlijst"] = 2019
        elif spot_type in [Spot.SpotType.protocol_ernstig, Spot.SpotType.protocol_dodelijk]:
            data["jaar_ongeval_quickscan"] = 2020

        response = self.write_client.post(url, data=data)
        self.assertStatusCode(url, response, expected_status=201)

        del data["point"]
        self.assertTrue(Spot.objects.filter(**data).exists())
```